### PR TITLE
Feature/http file extensions

### LINKED ISSUE #286: LD documents without content type not queryable.
#### Issue type:

- :bug: Bug

____
#### Description:

Documents without a Content-Type header value like http://wikiba.se/ontology#label are not queryable. We should provide a fallback-mechanism based on file extensions for such situations, just like we do for querying local files.

### PR DESCRIPTION
**Pull request by rubensworks**
This abstracts the fallback mechanism for determining media types based on file extensions, so that it can be used in the HTTP-based RDF dereferencer as well.

Closes #286

### CODE REVIEW

File: packages/actor-init-sparql-file/config/sets/rdf-source-identifiers.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "@context": [
+    "https://linkedsoftwaredependencies.org/bundles/npm/@comunica/actor-init-sparql/^1.0.0/components/context.jsonld",
```
Comment (joachimvh) [inline packages/actor-init-sparql-file/config/sets/rdf-source-identifiers.json]: Why was this added?

File: packages/actor-rdf-source-identifier-file-content-type/lib/ActorRdfSourceIdentifierFileContentType.ts [hunk 2]
```diff
@@ -28,9 +28,9 @@ export class ActorRdfSourceIdentifierFileContentType extends ActorRdfSourceIdent
     const httpAction: IActionHttp = { context: action.context,
       init: { headers, method: 'HEAD' }, input: action.sourceValue };
     const httpResponse: IActorHttpOutput = await this.mediatorHttp.mediate(httpAction);
-    if (!httpResponse.ok || !httpResponse.headers.has('Content-Type')
-      || !this.allowedMediaTypes.find((mediaType: string) => httpResponse.headers
-        .get('Content-Type').indexOf(mediaType) >= 0)) {
+    if (!httpResponse.ok || (httpResponse.headers.has('Content-Type')
+      && !this.allowedMediaTypes.find((mediaType: string) => httpResponse.headers
+        .get('Content-Type').indexOf(mediaType) >= 0))) {
```
Comment (joachimvh) [inline packages/actor-rdf-source-identifier-file-content-type/lib/ActorRdfSourceIdentifierFileContentType.ts]: What about files with no content type and no valid file extension?

File: packages/actor-init-sparql-file/config/sets/rdf-source-identifiers.json (hunk 1 above)
Comment (rubensworks) [reply packages/actor-init-sparql-file/config/sets/rdf-source-identifiers.json]: Because that's where the `config-sets` prefix is defined. This config did not work before apparently.

File: packages/actor-rdf-source-identifier-file-content-type/lib/ActorRdfSourceIdentifierFileContentType.ts (hunk 2 above)
Comment (rubensworks) [reply packages/actor-rdf-source-identifier-file-content-type/lib/ActorRdfSourceIdentifierFileContentType.ts]: Those will simply fail :-)

We'll need this for that: https://github.com/comunica/comunica/issues/289
